src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1409,68 +1481,101 @@ impl Cage {
                             }
                         }
 
+                        // For `CharDev` type inode, the reading happens from the Character Device
+                        // file, with each device type returning different results returned
+                        // from the `_read_chr_file` file.
                         Inode::CharDev(char_inode_obj) => {
+                            // reads from character devices by matching the device number (DevNo) of
+                            // the DeviceInode.
                             self._read_chr_file(&char_inode_obj, buf, count)
```

Comment:
How do we handle the result of the _read_chr_file function?

Reply:
Here, this function returns the number of bytes read into the buffer. And since for returning values in Rust, we just specify the return value, here we are just calling the function which returns the expected result. 
Will update this comment for better understanding.

Reply:
Updated the comments with the description of the return value.